.github/workflows/build-and-run.yml

```
@@ -0,0 +1,116 @@
+name: Build-And-Run Tests
+
+on: [pull_request, push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+jobs:
+  build_and_test:
+    env:
+      ALL_EXCLUDE: stage_dem|pybind\.unwrap\.phass
+
+
+    defaults:
```

Another very minor pedantic comment --

In general in this file, you use a single blank line to separate blocks of code. But here you're using 2 blank lines. I think it would look a little cleaner if we tried to consistently use the same number of empty lines to space things out

```suggestion
      ALL_EXCLUDE: stage_dem|pybind\.unwrap\.phass

    defaults:
```